packages/optimizers/terser/src/TerserOptimizer.js

```diff
@@ -0,0 +1,40 @@
+// @flow
+
+import nullthrows from 'nullthrows';
+import {minify} from 'terser';
+import {Optimizer} from '@parcel/plugin';
+import {loadConfig} from '@parcel/utils';
+
+export default new Optimizer({
+  async optimize({contents, bundle, options}) {
+    if (!options.minify) {
+      return contents;
+    }
+
+    if (typeof contents !== 'string') {
+      throw new Error(
+        'TerserOptimizer: Only string contents are currently supported'
```

Comment:
ahh this is interesting. maybe once we introduce ast sharing for optimizers, we could end up with a `getCode`, `getBuffer`, `getAST` etc. on a `Bundle` similar to the ones in `Asset`